tests/analysis/test_validators.py

```diff
@@ -76,6 +76,242 @@ def test_bmds3_partial(self):
         assert validators.validate_input(data, partial=True) is None
         assert validators.validate_input(data) is None
 
+    def test_nested_dichotomous(self):
+        data: Dict[str, Any] = {
+            "bmds_version": bmds.constants.BMDS330,
+            "dataset_type": bmds.constants.NESTED_DICHOTOMOUS,
+        }
+        print(data)
+        assert validators.validate_input(data, partial=True) is None
+
+        # but fails when complete
+        with pytest.raises(ValidationError) as err:
+            validators.validate_input(data)
+        _missing_field(err, "datasets")
+
+        # add datasets, try again
+        data["datasets"] = [
```

Comment:
can we make a fixture? using `# fmt: off` w/ formatting like javascript?  see other examples in bmds-`server\tests\conftest.py`

Reply:
moved dataset into pytest fixtures.